flycheck.el

```diff
@@ -6920,6 +6920,41 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+(defun flycheck-rebar3-project-root (&optional _checker)
+  "Return directory where =rebar.config= is located."
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
+                        rebarconfig)))
+    project-dir))
+
+(flycheck-define-checker erlang-rebar3
+  "An Erlang syntax checker using the rebar3 build tool."
+  :command ("rebar3" "compile")
+  :error-parser
+  (lambda (output checker buffer)
+    ;; rebar3 outputs ANSI terminal colors, which don't match up with
+    ;; :error-patterns, so we strip those color codes from the output
+    ;; here before passing it along to the default behavior. The
+    ;; relevant disucssion can be found
+    ;; https://github.com/flycheck/flycheck/pull/1144
+    (require 'ansi-color)
+    (flycheck-parse-with-patterns (ansi-color-filter-apply output) checker buffer))
```

Comment:
@joedevivo Thanks for the explanatory comment.

Now, to fix the compiler warning use the following as argument to `flycheck-parse-with-patterns`:

```elisp
(and (fboundp 'ansi-color-filter-apply) (ansi-color-filter-apply output))
```

The `fboundp` guard is entirely redundant, because we _know_ that `(require 'ansi-color)` brings that function in scope, but the byte compiler can't make that link.  However, the `fboundp` guard convinces it that we're only calling the function when it's defined and thus removes the warning.

Ugly, but works, and much evil is achieved 😎